core/threadpool.c

```diff
@@ -291,15 +290,15 @@ static void *threadpool_thread(void *threadpool)
         pool->count -= 1;
 
         /* Unlock */
-        pthread_mutex_unlock(&(pool->lock));
+        cthreads_mutex_unlock(&(pool->lock));
 
         /* Get to work */
         (*(task.function))(task.argument);
     }
 
     pool->started--;
 
-    pthread_mutex_unlock(&(pool->lock));
-    pthread_exit(NULL);
+    cthreads_mutex_unlock(&(pool->lock));
+    cthreads_thread_close(NULL);
     return(NULL);
-}
+}
```

Comment:
I think modelling it after the UNIX threading API was a good decision; UNIX is the common-denominator OS, so, it makes sense to follow their lead.